server/conf.go

```diff
@@ -475,14 +476,17 @@ func ConfigureOptions(fs *flag.FlagSet, args []string, printVersion, printHelp,
 	}
 
 	if len(stanClusterPeers) > 0 {
-		sopts.ClusterPeers = strings.Split(stanClusterPeers, ",")
-		for i, peer := range sopts.ClusterPeers {
-			sopts.ClusterPeers[i] = strings.TrimSpace(peer)
+		sopts.Clustering.Peers = strings.Split(stanClusterPeers, ",")
+		for i, peer := range sopts.Clustering.Peers {
+			sopts.Clustering.Peers[i] = strings.TrimSpace(peer)
 		}
+
+		// If clustered, override store sync configuration with cluster sync.
```

Comment:
I would move this into server code, before creating the Store. There are users that may run server from code by providing options without invoking this config options.

Reply:
It will probably apply to all processing you do here with cluster related options. Keep in mind that user *can* start the server from code using `RunServerWithOpts()`.